README.md

```diff
@@ -0,0 +1,143 @@
+6. module checks URL fragment and cache for access_token (found)
+7. module contacts The Key (casTicketPath) using access_token: access_token(The Key) and service: managedApi being requested
+8. Key returns a ticket to module
+9. module uses ticket to authenticate with auth-api(authenticationApiBaseUrl)
+10. auth-api returns access_token(service) to module
+11. original request sent with access_token(rails) to service
+
+---
+
+## Installation
+
+Choose your preferred method:
+
+* Bower: `bower install angular-cas-auth-api`
+* NPM: `npm install --save angular-cas-auth-api`
+* Download: [angular-cas-auth-api](https://raw.github.com/CruGlobal/angular-cas-auth-api/master/dist/cas-auth-api.min.js)
+
+## Usage
+
+###### 1. Request Client ID (optional)
+If you are going to use The Key (Cru CAS) as your CAS authenticator you will need to get a Client ID.
+send an email to [apps@cru.org](mailto:apps@cru.org?subject=New OAuth Client Request for The Key) with the subject **New OAuth Client Request for The Key**.
+
+
+###### 2. Configure `casAuthApi`:
+
+```js
+angular.module('myApp', ['cas-auth-api'])
+  .config(['casAuthApiProvider', function(casAuthApiProvider) {
+    casAuthApiProvider.configure({
+        requireAccessToken: true,
+        cacheAccessToken: true, // If true you must include lscache #optional
+        redirectUrl: 'http://localhost/', // URL to redirect to after OAuth Authentication #required
+        clientId: '1234567890', // Client ID related to redirect URL #required
+        oAuth: true // #required
+    });
+  }]);
+```
+
+###### 3. Catch `casAuthApi` errors and do something with them (optional):
+
+```js
+angular.module('myApp', ['cas-auth-api'])
+  .run(['$rootScope', '$window', function($rootScope, $window) {
+    $rootScope.$on('cas-auth-api:error', function(event, rejection) {
+      if ('ERR_SERVICE' === rejection.code) {
+        // retrieving service object failed
+      }
+
+      if ('ERR_TICKET' === rejection.code) {
+        // retrieving ticket object failed
+      }
+
+      if ('ERR_TOKEN' === rejection.code) {
+        // retrieving token object failed
+      }
+
+      // Redirect to `/login` with the `error_reason`.
+      return $window.location.href = '/login?error_reason=' + rejection.message;
+    });
+  }]);
+```
+
+**NOTE**: An *event* `cas-auth-api:error` will be sent every time a `responseError` is emitted:
+
+* `{ code: 'ERR_SERVICE', message: 'Failed to fetch service.'}`
+* `{ code: 'ERR_TICKET', message: 'Failed to fetch ticket.'}`
+* `{ code: 'ERR_TOKEN', message: 'Failed to fetch token.'}`
+
+## API
+
+#### CasAuthApiProvider
+
+Configuration defaults:
+
+```js
+casAuthApiProvider.configure({
+  endpoints: {
+      'service': 'service',
+      'token': 'token/new'
+  },
+  ticketUrl: '',
+  maxAttempts: 3,
+  requireAccessToken: false,
+  cacheAccessToken: false,
+  cacheExpiresMinutes: 25,
+  managedApis: [],
+  casBaseUrl: 'https://thekey.me',
```

Comment:
I would probably use these defaults:
```
casBaseUrl: 'https://thekey.me/cas/' ,
casLoginPath: 'login',
casTicketPath: 'api/oauth/ticket',
```
it's a java webapp that happens to be currently deployed in the `cas` subdirectory.